### ISSUE TITLE
Possible to add Node.js 12 support to v4?

### ISSUE THREAD

**Issue by MattMorgis** [Level/leveldown#641]
Hi,

I understand `leveldown@v5` is the minimum version for Node.js 12 support. 

Is it feasible to add Node.js 12 support to v4? I'd be happy to help out, but curious if it's a lost cause from the start. 

I ask because I am working on updating `level-party`. I was able to get it to run on Node.js 10 and 11 by having it use a combination of `levelup@1.x` and `leveldown@4.x` (from `leveldown@1.x`. 

It seems a refactor to `leveldown@5x` would require a refactor to `abstract-leveldown@v6` and `levelup@2.x+`, which has some changes to encodings. If it's possible to add Node.js 12 support to v4, I could avoid the big reactor for now.

**Comment by vweevers** [comment]
> I ask because I am working on updating level-party

What doesn't work when you combine `level-party` with `leveldown@5`?

I prefer fixing that over maintaining two release lines of `leveldown`.

**Comment by MattMorgis** [comment]
>I prefer fixing that over maintaining two release lines of leveldown.

Totally understand. I am new to all of these level modules, and am just trying to be a good open source citizen and update level-party as it failed to install when we upgraded to Node.js 10. I will take any guidance on how to upgrade it properly.  

I have a fork of `level-party` here.

If you change this to `"1.8.0"`, it will work on Node.js 10 and 11 (`levelup@1.x` and `levelup@2.x`), but won't install on Node.js 12.
https://github.com/MattMorgis/level-party/blob/node-10-support/package.json#L7

If you change it to `"1.8.1"` (`levelup@1.x` and `leveldown@5.x`), it installs on Node.js 12, but `npm test` will fail across all versions. 
https://github.com/MattMorgis/level-party/blob/node-10-support/package.json#L7

The culprit is this test, specifically this line not getting called. I've gone down many rabbit holes trying to figure out why, but none have seemed to lead to the answer so far. 
https://github.com/MattMorgis/level-party/blob/master/test/bytewise.js#L31

**Comment by vweevers** [comment]
I think the first step is to update https://github.com/mafintosh/multileveldown. Then we might need not even need API changes in `level-party`, which is just a thin wrapper around `multileveldown`.

**Comment by MattMorgis** [comment]
I've been down that rabbit hole. I have my own fork [here](https://github.com/MattMorgis/multileveldown/tree/node-10-support), which was from [this fork](https://github.com/jasnell/multileveldown/tree/fix-buffer-constructor).

It seems when upgrading `multileveldown` to use `levelup@2.x+`, and bringing `encoding-down` into the mix breaks a lot there and which bubbles up into `level-party`

I can submit something that reproduces it, but the main tests that broke seemed to be almost there. It would want int `64`, but instead would get the Buffer of that data or string `"64"`.

**Comment by vweevers** [comment]
Yeah, I had a look at `multileveldown` just now. With a few tweaks, gets and puts and dels are working, read streams are not. I can take a closer look at the end of the week.

I'm afraid that rabbit hole (which I agree it is!) is the only proper way to fix this.

/cc @mafintosh Are you still (interested in) maintaining `multileveldown`? If not, we could consider moving it. But maybe not to Level at first, there's enough work here.

**Comment by vweevers** [comment]
OK, got all `multileveldown` tests to pass. But to make it work @ralphtheninja I'm afraid we're gonna have to revert this revert: https://github.com/Level/codec/pull/23 (I will attempt to explain later)

**Comment by vweevers** [comment]
https://github.com/vweevers/multileveldown/tree/wip-upgrade
